lib/util/import-file.js

```diff
@@ -5,7 +5,7 @@
  * @todo WARN on version 10 and '--experimental-modules' and '--esm'
  */
 module.exports = function importFile(filepath) {
-  return filepath.endsWith('.mjs')
+  return filepath.endsWith('.mjs') || process.env.npm_package_type === 'module'
```

Comment:
Could you please read `process.env.npm_package_type` value once and cache it? `process.env` calls are expensive (https://www.reddit.com/r/node/comments/7thtlv/performance_penalty_of_accessing_env_variables/)

Reply:
Also I would swap the order of checks, if it's set per-process, no need to check filepath.

Reply:
Yes, of course. I wasn't aware it's expensive, good to know.

Maybe there's a better way to check for esm?

Reply:
If there is, I'm not aware of it, unfortunately. I think this solution will do nicely for now. How urgently do you need this change published, btw?

Reply:
Not urgent, take your time.

It's for a personal project. I can always use [patch-package](https://github.com/ds300/patch-package) just in case.

Reply:
Gotcha. We plan to do the final release in 0.21 line next week, so this one will be included.

Reply:
Sounds good. Thanks!